File: shell/ubuntu_release.go
package main

import (
	"fmt"
	"log"
	"os"
	"os/exec"
)

func main() {
	var ubuntuRelease *exec.Cmd = exec.Command(
		"sudo",
		"do-release-upgrade",
	)

	var ubuntuReleaseError error = ubuntuRelease.Run()

	if ubuntuReleaseError != nil {
		log.Fatalln("Ubuntu Release Upgrade Failed: \n", ubuntuReleaseError)
	}

	fmt.Println("Running Ubuntu Release Upgrade: \n", ubuntuRelease)

	os.Exit(0)
}

File: upt/src/upgrades/ubuntu_release.rs
use std::{
    io::{Error, StdoutLock, Write, stdout},
    process::{Command, Output, exit},
    result::{
        Result,
        Result::{Err, Ok},
    },
};

// Ubuntu System Release Upgrade
pub fn release_ubuntu() -> () {
    let ubuntu_release: Result<Output, Error> = Command::new("do-release-upgrade").output();
    let mut standard_output: StdoutLock = stdout().lock();

    match ubuntu_release {
        Ok(release) => {
            standard_output.write_all(&release.stdout).unwrap();
            writeln!(standard_output, "{}", release.status).unwrap();
        }
        Err(error) => {
            eprintln!("Error Executing Ubuntu System Release Upgrade: {}", error);
            exit(1);
        }
    };

    return ();
}
